File: problems/1318_minimum-flips-to-make-a-or-b-equal-to-c/main.go
// +build 1712

package main

func minFlips(a int, b int, c int) int {
	var result int

	for c > 0 {
		switch c & 1 {
		case 1:
			if a&1 == 0 && b&1 == 0 {
				result++
			}

		case 0:
			result += a&1 + b&1
		}

		a = a >> 1
		b = b >> 1
		c = c >> 1
	}

	for a > 0 {
		result += a & 1
		a = a >> 1
	}
	for b > 0 {
		result += b & 1
		b = b >> 1
	}

	return result
}

func main() {
	println(minFlips(2, 6, 5))
	println(minFlips(4, 2, 7))
	println(minFlips(1, 2, 3))
	println(minFlips(8, 3, 5))
}

File: problems/1318_minimum-flips-to-make-a-or-b-equal-to-c/__main__.py
import unittest


class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        ret = 0
        while a != 0 or b != 0 or c != 0:
            if c & 1 == 1:
                ret += 1 if a & 1 == 0 and b & 1 == 0 else 0
            else:
                ret += (a & 1) + (b & 1)
            a = a >> 1
            b = b >> 1
            c = c >> 1
        return ret


class TestSolution(unittest.TestCase):
    def test_minFlips(self):
        self.assertEqual(Solution().minFlips(2, 6, 5), 3)
        self.assertEqual(Solution().minFlips(4, 2, 7), 1)
        self.assertEqual(Solution().minFlips(8, 3, 5), 3)
